bin/cdklocal

```diff
@@ -204,34 +205,55 @@ const patchCurrentAccount = (SDK) => {
 };
 
 const patchToolkitInfo = (ToolkitInfo) => {
-  const {
-    BUCKET_NAME_OUTPUT, BUCKET_DOMAIN_NAME_OUTPUT
-  } = require("aws-cdk/lib/api/bootstrap/bootstrap-props");
-
-  const setBucketUrl = function setBucketUrl(object) {
+  const setBucketUrl = function setBucketUrl(object, bucket, domain) {
+    const newBucketUrl = `https://${domain.replace(`${bucket}.`, "")}:${EDGE_PORT}/${bucket}`;
+    if (DEBUG_ON) {
+      console.debug("Old bucketUrl:", object.bucketUrl);
+      console.debug("New bucketUrl:", newBucketUrl);
+    }
     Object.defineProperty(object, "bucketUrl", {
-      async get() {
-        const bucket = this.requireOutput(BUCKET_NAME_OUTPUT);
-        const domain = this.requireOutput(BUCKET_DOMAIN_NAME_OUTPUT) || await getLocalHost();
-        return `https://${domain.replace(`${bucket}.`, "")}:${EDGE_PORT}/${bucket}`;
+      get() {
+        return newBucketUrl;
       }
     });
   };
 
+  // Pre-fetch the necessary values for the bucket URL
+  const prefetchBucketUrl = async (object) => {
+    // Has been observed that the object is not always an instance of ToolkitInfo
+    if (object != {}) {
```

Comment:
Where does the comparison with `{}` come from here? It doesn't seem to align with the comment above which could be checked directly or we could try to explicitly check for the availability of the bucketname/url properties on the object. before manipulating them.

Reply:
We can do a simple `if (object)` instead of also checking the presence of the individual attributes. Accessing the individual attributes can lead to raised exceptions:

https://github.com/aws/aws-cdk/blob/87e21d625af86873716734dd5568940d41096c45/packages/aws-cdk/lib/api/toolkit-info.ts#L190-L196:
```javascript
  public get bucketUrl(): string {
    throw new Error(this.errorMessage);
  }

  public get bucketName(): string {
    throw new Error(this.errorMessage);
  }
```

Instead, we're gonna be using the `hasOwnProperty` method for that.